### PR TITLE
AF-2759 V2 Prep: deprecate docs, examples, saucelabs; add dart1-only and dart2-only

### PR DESCRIPTION
**Pull request by evanweible-wf**
## Description
In preparation for the 2.0.0 release of dart_dev:

* Deprecate `docs` task
* Deprecate `examples` task
* Deprecate `saucelabs` task
* Add `dart1-only` conditional task
* Add `dart2-only` conditional task
* Update/add relevant documentation

## Testing
- [ ] CI passes (tests added)

## Code Review
@Workiva/app-frameworks

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on HipChat: InfoSec Forum._

File: test/integration/dart_x_only_test.dart [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+      test('is skipped', () async {
+        final result = await runDart2Only(['coverage']);
+        expect(result.skipped, isTrue);
+        expect(result.exitCode, 0);
+      });
+    });
+  }, tags: 'dart1-only');
+
+// TODO
```
[the remark this answers is not in the document]
Comment (evanweible-wf) [reply test/integration/dart_x_only_test.dart]: For later, I have it added here: https://github.com/Workiva/dart_dev/pull/274

File: test/integration/dart_x_only_test.dart [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    run('dart2-only', targetArgs);
+
+Future<Result> run(String task, Iterable<String> targetArgs) async {
+  await Process.run('pub', ['get'], workingDirectory: projectPath);
+
+  var args = ['run', 'dart_dev', task]..addAll(targetArgs);
+  TaskProcess process =
+      new TaskProcess('pub', args, workingDirectory: projectPath);
+  final stdout = await process.stdout.join('\n');
+  print(stdout);
```
Comment (corwinsheahan-wf) [inline test/integration/dart_x_only_test.dart]: want to leave this in here?

Comment (evanweible-wf) [reply test/integration/dart_x_only_test.dart]: Yeah, it's helpful for debugging and sanity checking that they're actually doing what we expect

**Comment by corwinsheahan-wf** [comment]
QA +1

- [x] Testing instruction
- [x] Dev +1's
- [x] [Dev/QA +10 with detail of what was tested](https://github.com/Workiva/dart_dev/pull/275#issuecomment-427622742)
- [x] Security review (if required)
- [x] Unit tests created/updated
- [x] All unit tests pass
- [x] Rosie has run and reports properly the release the ticket will be included in

Merging into master.
@Workiva/release-management-p

**Pull request merged**